projects/plugins/jetpack/_inc/lib/core-api/wpcom-endpoints/class-wpcom-rest-api-v2-endpoint-coauthor.php

```diff
@@ -0,0 +1,105 @@
+<?php
+/**
+ * REST API endpoint for the CoAuthor blocks.
+ *
+ * @package automattic/jetpack
+ * @since $$next-version$$
+ */
+
+/**
+ * Class WPCOM_REST_API_V2_Endpoint_Coauthor
+ */
+class WPCOM_REST_API_V2_Endpoint_Coauthor extends WP_REST_Controller {
+	/**
+	 * Namespace prefix.
+	 *
+	 * @var string
+	 */
+	public $namespace = 'wpcom/v2';
+
+	/**
+	 * Endpoint base route.
+	 *
+	 * @var string
+	 */
+	public $rest_base = 'coauthor';
+
+	/**
+	 * WPCOM_REST_API_V2_Endpoint_Coauthor constructor.
+	 */
+	public function __construct() {
+		$this->is_wpcom = false;
+
+		if ( defined( 'IS_WPCOM' ) && IS_WPCOM ) {
+			$this->is_wpcom = true;
```

Comment:
@jeherve This code being in `core-api/wpcom-endpoints`, I was expecting is_wpcom to be always true. Is that loaded in all context as well?!

Reply:
It is loaded in all contexts, yes. The wpcom/v2 endpoints are available on self-hosted sites as well. See https://jeremy.hu/wp-json/wpcom/v2/ for example.

Reply:
What is their purpose outside of WPCOM? They somehow forward the request top WPCOM?

Reply:
Ans I have to listen to ["La rue ketanou"](https://jeremy.hu/music-les-cigales-la-rue-ketanou/) now...

Reply:
> They somehow forward the request top WPCOM?

Exactly.

> I have to listen to ["La rue ketanou"](https://jeremy.hu/music-les-cigales-la-rue-ketanou/) now...

:) 🪗